### DISCUSSION TITLE
Backward Projection with Metashape Data on ver2.0

### DISCUSSION THREAD

**Discussion by Ken-Kuroki** [UTokyo-FieldPhenomics-Lab/EasyIDP#33] [Q&A] [answered]
Hi, I've been trying out ver2.0 and it is much easier to use with Pix4D projects. Great job!
But I'm having an issue with Metashape projects. Following is the minimum code required to replicate it with your test dataset.

```.py
import easyidp as idp
project_path = "/data/2017_tanashi_lotus/170531.Lotus.psx"
dsm_path = "/data/2017_tanashi_lotus/170531.Lotus.outputs/170531.Lotus_dsm.tif"
ms = idp.Metashape(project_path, chunk_id=0)
roi = idp.ROI("/data/2017_tanashi_lotus/plots.shp")
roi.get_z_from_dsm(dsm_path)
out = ms.back2raw(roi)
```

The resulting `out` contains only `nan` for each image and plot. This may be totally off the point but I noticed `ms.crs` is set to 4326 as opposed to 32654. I changed it to 32654 by `pyproj.CRS.from_user_input("epsg:32654")` and then `out` became not `nan` but instead coordinates that probably work. Advise me if I'm using it wrongly.

**Comment by HowcanoeWang** [comment]
yes, you are right, currently the metashape have not been fully implemented. You need to specify ms.crs to dom's crs to get correct results. It will be fixed soon.

**Reply by HowcanoeWang** [reply]
open the issue #34

**Reply by Ken-Kuroki** [reply]
Thanks a lot! It's not a big deal to specify CRS manually but would surely be nice if that can be done automatically.

**Comment by HowcanoeWang** [comment, marked as answer]
@Ken-Kuroki Should support now in `v2.0.0.dev3`

```python
# optional
ms.crs = pyproj.CRS.from_epsg(32654)

# it will read crs.roi directly
ms.back2raw(roi)
```

It should produce the correct results.

**Reply by Ken-Kuroki** [reply]
Thank you so much for addressing it! I'll try it out!
